TLM/TLM/Manager/Impl/RoutingManager.cs

```diff
@@ -2342,5 +2346,45 @@ bool laneChangesAllowed
 
             RecalculateAll();
         }
+
+        /// <summary>
+        /// if the direction of the given lanes do not match then similar lane index must be reversed.
+        /// </summary>
+        private bool ShouldReverseSimilarLaneIndex(
+            bool segmentInvert1, NetInfo.Lane laneInfo1, bool startNode1,
+            bool segmentInvert2, NetInfo.Lane laneInfo2, bool startNode2) {
+#if DEBUG
+            bool logRouting = DebugSwitch.Routing.Get(); 
+#else
+            const bool logRouting = false;
+#endif
+            bool both1 = laneInfo1.m_finalDirection == NetInfo.Direction.Both;
+            bool both2 = laneInfo2.m_finalDirection == NetInfo.Direction.Both;
+            bool backward1 = laneInfo1.IsGoingBackward();
+            bool backward2 = laneInfo2.IsGoingBackward();
```

Comment:
Move in to `else` block? We don't need to know `backwardN` until we reach the `else` block?

Note: I don't know if the compiled IL will do that optimisation on its own.

Reply:
it would be useful for the log.
If you look at the rest of the code in routing manager, the scope of the variables is increased for the sake of debug log.
at the cost of a bit of performance. but this code is not that performance critical so its fine (performance matters but only in terms of reducing loading time).